src/Dumplie/Domain/Customer/Product.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Domain\Customer;
+
+use Dumplie\Domain\Customer\Exception\InvalidArgumentException;
+use Dumplie\Domain\SharedKernel\Money\Price;
+
+final class Product
+{
+    /**
+     * @var string
+     */
+    private $sku;
+
+    /**
+     * @var Price
+     */
+    private $price;
+
+    /**
+     * @var bool
+     */
+    private $available;
```

Comment:
unavailable product can still be bought and paid for...
Actually it strongly depends on business politics - eg. Amazon allows you to buy unavailable products but will request payment only when they become available.
Another case is - shops that promote some sort of membership and allow customers of special type to buy products before they go _public_.

Ergo: availability should be under some kind of interface with basic boolean implementation.